model/task/task.go

```diff
@@ -1313,6 +1313,23 @@ func FindMidwayTask(t1, t2 Task) (*Task, error) {
 	return FindOne(db.Query(ByRevisionOrderNumber(t1.BuildVariant, t1.DisplayName, t1.Project, t1.Requester, mid)))
 }
 
+// FindMidwayTaskFromIds gets the task between two task given that they are
+// from the same project, requester, build variant, and display name. The
+// order of the ID's does not matter and if the task passed cannot have a
+// middle (i.e. it is sequential tasks or the same task) it will return the
+// the first task given.
+func FindMidwayTaskFromIds(t1Id, t2Id string) (*Task, error) {
+	t1, err := FindOneId(t1Id)
+	if err != nil || t1 == nil {
```

Comment:
the task being nil and finding an error looking it up from the DB are two separate error cases, so we should treat them as such with separate err messages, so it's best to separate them. Here's an [example](https://gist.github.com/hadjri/cc38c4ed5758423c54714adca823ce78#file-task_lifecycle-go-L130-L136)